Add operations for FreezeUnit and ThawUnit

irishair opened the pull request:
We are trying to use bluechi for multi-node container ochestration in our automotive HPC.
In our system, we have additional requirements for suspend and resume of units.
From systemd 246, it supports "systemctl freeze/thaw" and D-Bus APIs for those based on cgroup v2.
So we added a new feature that supports freeze and thaw units on multi-node to bluechi, and we would like to submit this patch.

This PR contains following changes.
- Add FreezeUnit/ThawUnit operations to agent and manager
- Add freeze/thaw commands to client
- Add FreezeUnit/ThawUnit interfaces to DBus-API

Example test results
```
# bluechictl freeze ak7_master_sub v2x-client.service
Unit v2x-client.service freeze operation done


# systemctl status v2x-client.service
● v2x-client.service - systemd service for v2x client container
Loaded: loaded (/etc/containers/systemd/v2x-client.container; generated)
Active: active (running) (frozen) since Wed 2023-08-21 14:26:34 KST; 2min 37s ago
Main PID: 3451 (conmon)
  Tasks: 21 (limit: 7782)
Memory: 25.8M
    CPU: 6.107s
CGroup: /system.slice/v2x-client.service
         ├─libpod-payload-7ea31bcb6eb869693e09d7ca49b57cfbb3cbf51adfdaa0c54cc008c5068119a4
         │ ├─3454 /bin/bash /usr/local/akitos/communications/v2xservice/bin/v2x_client.sh start_sync
         │ └─3496 /usr/lib/v2x_agent/v2x_agent --config=/tmp/v2xservice/conf/v2x_config.json
         └─runtime
           └─3451 /usr/bin/conmon --api-version 1 -c 7ea31bcb6eb869693e09d7ca49b57cfbb3cbf51adfdaa0c54cc008c5068119a4 -u 7ea31bcb6eb869693e09d7ca49b57cfbb3cbf51adfdaa0c54cc008c5068119a4 -r /usr/bin/crun>


# bluechictl thaw ak7_master_sub v2x-client.service
Unit v2x-client.service thaw operation done
```

engelmi reviewed (approved):
LGTM, great work!

@irishair Could you fetch the latest changes from main? Then the 2 failing integration tests should pass as well and we could merge. 

It would be great to have integration tests for these new features, but that can be done in a separate PR - I'll create an issue for it. 
**Update:** Created https://github.com/containers/bluechi/issues/495

irishair committed:
Add operations for FreezeUnit and ThawUnit
- Add FreezeUnit/ThawUnit operations to agent and manager
- Add freeze/thaw commands to client
- Add FreezeUnit/ThawUnit interfaces to DBus-API

Signed-off-by: minsoo.kim <minsoo.kim@42dot.ai>

mwperina reviewed (approved):
LGTM
